// protoc --python_out=design/archiving --proto_path=protobuf protobuf/archiving.proto

package archiving;

message Field {
	required string field_name = 1;
	required string value_type = 2;
	repeated string labels = 3;
}

message Record {
	required uint32 record_id = 1;
	required string record_name = 2;
	repeated Field fields = 3;
}

message Header {
	required string name = 1;
	required uint32 timestamp_origin = 2; // UTC seconds since the UNIX epoch
	repeated Record records = 3;
}

message Value {
	optional double double = 1;
	optional uint32 unsigned = 2;
	optional sint32 signed = 3;
}

message Update {
	required uint32 record_id = 1;
	optional uint32 elapsed_days = 2; // since timestamp_origin
	required uint32 elapsed_seconds = 3;
	optional uint32 elapsed_millisecs = 4;
	repeated Value values = 5;
}
